lib/AST/Bounds.cpp

```diff
@@ -905,6 +905,17 @@ class DeclBoundsFinder : public clang::DeclVisitor<DeclBoundsFinder>,
     return meth->getParent()->isLambda();
   }
 
+  static bool IsImplicitMethodInLambda(clang::FunctionDecl *decl) {
+    auto meth = clang::dyn_cast<clang::CXXMethodDecl>(decl);
+    if (!meth) {
+      return false;
+    }
+    auto lambdaClass = clang::dyn_cast<clang::CXXRecordDecl>(
```

Comment:
Use `meth->getParent()` instead, which gives you a `CXXRecordDecl *`.

Reply:
Also, perhaps re-implement as `return decl->isImplicit() && IsMethodInLambda(decl);`?